openmc/statepoint.py

```diff
@@ -534,8 +541,16 @@ def get_tally(self, scores=[], filters=[], nuclides=[],
 
                 # Iterate over the Filters requested by the user
                 for filter in filters:
-                    if filter not in test_tally.filters:
-                        contains_filters = False
+                    contains_filters = False
+
+                    # Test if requested filter is a subset of any of the test
+                    # tally's filters and if so continue to next filter
+                    for test_filter in test_tally.filters:
+                        if test_filter.is_subset(filter):
+                            contains_filters = True
+                            break
+
+                    if not contains_filters:
                         break
```

Comment:
I'm not opposed to the use of the ``for/else`` construct, but in this scenario I think it will lead to the wrong behavior. In particular, your code snippet will fail since it will not check all filters before calling `break`, and hence it may return a `Tally` which only has one but not necessarily all of the requested filters. The conditionals in this code block and others in ``StatePoint.get_tally(...)`` call ``continue`` when the loop(s) are terminated early (*i.e.*, execution hits ``break``). Since the `else` in `for/else` is only executed for normal loop termination, this construct can't be used here (at least based on my understanding).